modules/assim.batch/R/pda.define.llik.R

```diff
@@ -0,0 +1,34 @@
+##' Define PDA Likelihood Functions
+##'
+##' @title Define PDA Likelihood Functions
+##' @param all params are the identically named variables in pda.mcmc / pda.emulator
+##'
+##' @return List of likelihood functions, one for each dataset to be assimilated against.
+##'
+##' @author Ryan Kelly
+##' @export
+pda.define.llik.fn <- function(settings) {
+  # Currently just returns a single likelihood, assuming the data are flux NEE.
+  llik.fn <- list()
+  for(i in 1:length(settings$assim.batch$inputs)) {
+    # NEE + heteroskedastic Laplace likelihood
+    if(settings$assim.batch$inputs[[i]]$variable.id == 297 && 
+       settings$assim.batch$inputs[[i]]$likelihood == "Laplace") {
+        llik.fn[[i]] <- function(NEEm, obs) {
+          NEE.resid <- abs(NEEm - obs$NEEo)
+          NEE.pos <- (NEEm >= 0)
+          LL <- c(dexp(NEE.resid[NEE.pos], 1/(obs$b0 + obs$bp*NEEm[NEE.pos]), log=TRUE), 
+                  dexp(NEE.resid[!NEE.pos],1/(obs$b0 + obs$bn*NEEm[!NEE.pos]),log=TRUE))
+          return(list(LL=sum(LL,na.rm=TRUE), n=sum(!is.na(LL))))
```

Comment:
If we're looking over inputs on line 13, it doesn't make sense to have returns within that loop, since the function will never get to the second data set